Managers/ConsoleMan.cpp

```diff
@@ -1,541 +1,423 @@
-            // Add semicolon so the line input becomes a statement
-            if (!itr->empty() && (*itr)[itr->length() - 1] != ';')
-                writer << ";";
-            writer << "\n";
-        }
-
-        PrintString("SYSTEM: Console input log saved to " + filePath);
-    }
-}
-
-
-//////////////////////////////////////////////////////////////////////////////////////////
-// Method:          SaveAllText
-//////////////////////////////////////////////////////////////////////////////////////////
-// Description:     Writes the entire console buffer to a file.
-
-void ConsoleMan::SaveAllText(std::string filePath)
-{
-    Writer writer(filePath.c_str());
-
-    if (writer.WriterOK())
-    {
-        writer << m_pConsoleText->GetText();
-        PrintString("SYSTEM: Entire console contents saved to " + filePath);
-    }
-}
-
-//////////////////////////////////////////////////////////////////////////////////////////
-// Method:          ForceVisibility
-//////////////////////////////////////////////////////////////////////////////////////////
-// Description:     Sets console visible no matter what.
-// Arguments:       None.
-// Return value:    None.
-
-void ConsoleMan::ForceVisibility(bool visible)
-{
-	m_ForceVisible = visible; 
-	if (m_ForceVisible)
-	{
-		m_pParentBox->SetPositionAbs(0, 0);
+	const std::string ConsoleMan::c_ClassName = "ConsoleMan";
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	void ConsoleMan::Clear() {
+		m_ConsoleState = DISABLED;
+		m_ReadOnly = false;
+		m_ConsoleScreenRatio = 0.3F;
+		m_GUIScreen = 0;
+		m_GUIInput = 0;
+		m_GUIController = 0;
+		m_ParentBox = 0;
+		m_ConsoleText = 0;
+		m_InputTextBox = 0;
+		m_InputLog.clear();
+		m_InputLogPosition = m_InputLog.begin();
+		m_LastInputString.clear();
+		m_LastLogMove = 0;
 	}
-	else
-	{
-		m_pParentBox->SetPositionAbs(0, -m_pParentBox->GetHeight());
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	int ConsoleMan::Create() {
+		if (!m_GUIScreen) { m_GUIScreen = new AllegroScreen(g_FrameMan.GetBackBuffer32()); }
+		if (!m_GUIInput) { m_GUIInput = new AllegroInput(-1); }
+		if (!m_GUIController) { m_GUIController = new GUIControlManager(); }
+
+		if (!m_GUIController->Create(m_GUIScreen, m_GUIInput, "Base.rte/GUIs/Skins/MainMenu", "ConsoleSkin.ini")) {
+			RTEAbort("Failed to create GUI Control Manager and load it from Base.rte/GUIs/Skins/MainMenu/ConsoleSkin.ini");
+		}
+
+		m_GUIController->Load("Base.rte/GUIs/ConsoleGUI.ini");
+		m_GUIController->EnableMouse(false);
+
+		// Stretch the invisible root box to fill the screen
+		dynamic_cast<GUICollectionBox *>(m_GUIController->GetControl("base"))->SetSize(g_FrameMan.GetResX(), g_FrameMan.GetResY());
+
+		// Make sure we have convenient points to the containing GUI collection boxes that we will manipulate the positions of
```

Comment:
I assume `points` here is supposed to be `pointers`? Also, I don't think comment really explains anything, I'm not even sure what it's saying? And it doesn't explain the SetDrawType thing anyway.